source/common/common/hash.h

```diff
@@ -37,4 +39,16 @@ class HashUtil {
   }
 };
 
+struct CharStarHash {
+  size_t operator()(const char* a) const { return HashUtil::xxHash64(a); }
+};
+
+struct CharStarEqual {
+  size_t operator()(const char* a, const char* b) const { return strcmp(a, b) == 0; }
```

Comment:
No, we don't have the length of this string stored separately, so a simple strcmp what we need. We don't store the length because we are relying on nul-termination, and don't want to spend the extra 8 bytes per stat map entry on an aligned size_t.